x-pack/plugins/apm/public/hooks/use_environments_fetcher.tsx

```diff
@@ -29,11 +32,29 @@ export function useEnvironmentsFetcher({
               serviceName,
             },
           },
+        }).then((response) => {
+          // Spreading of original response is needed in order to display the api call in the inspector
+          return { ...response, environments: response.environments };
```

Comment:
I don't think we should modify our code to satisfy the inspector. I think a better fix is to call `addInspectorRequest` inside `createAutoAbortedAPMClient`:

https://github.com/elastic/kibana/blob/56d53675d164dd5110b627744fcdbe21e07188f7/x-pack/plugins/apm/public/hooks/use_fetcher.tsx#L56-L65

Reply:
like:
```
const promise = fn(createAutoAbortedAPMClient(signal, addInspectorRequest));
```

Reply:
I'd suggest just returning the environments:

```suggestion
          return response.environments;
```

Reply:
> I don't think we should modify our code to satisfy the inspector. I think a better fix is to call `addInspectorRequest` inside `createAutoAbortedAPMClient`:

@sqren Updated the `createAutoAbortedAPMClient` function to handle `inspector` logic. 
Regarding this comment

> I'd suggest just returning the environments:

I am no more modifying the returned value of `'GET /internal/apm/environments'` as its not required now.